server/data/prisonApiClient.ts

```diff
@@ -29,15 +37,140 @@ export default class PrisonApiClient extends AbstractHmppsRestClient {
     return this.put({ path: '/api/users/me/activeCaseLoad', data, authToken: user.token })
   }
 
-  async searchActivityLocations(
-    prisonCode: string,
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  async searchActivityLocations(prisonCode: string, date: string, period: string, user: ServiceUser): Promise<any[]> {
+    return this.get({
+      path: `/api/agencies/${prisonCode}/eventLocationsBooked`,
+      query: { bookedOnDay: date, timeSlot: period },
+      authToken: user.token,
+    })
+  }
+
+  async getActivitiesAtLocation(
+    locationId: string,
     date: string,
     period: string,
+    includeSuspended: boolean,
     user: ServiceUser,
-  ): Promise<Location[]> {
+    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  ): Promise<any[]> {
```

Comment:
This should be `Promise<ScheduledEvent[]>`

Reply:
As with the others - this types doesnt describe the actual response from the api. Not sure what to do here - I guess i could change the api test response fixtures to satisfy the type, other option is to create a new type but that seems just as wrong. Correct course of action might be to ask DPS to fix their api schema.

Reply:
ScheduledEvent doesnt have an offenderNo property which this service is returning - closer to PrisonerSchedule if anything

Reply:
As discussed yesterday i have created versions of the api types with very minor changes that are slightly more lenient - making a couple of attributes optional and changing enum types to strings. This has allowed me to replace the 'any' types.